doc/python/performance.md

```diff
@@ -0,0 +1,311 @@
+- When working with Plotly Express, pandas DataFrame, Polars DataFrame and PyArrow DataFrame objects passed to the `data_frame` argument of `px` functions.
+- Array objects that can be converted to `numpy.ndarray` objects, i.e., they implement `"__array__"` or `"__array_interface__"` and return a `numpy.ndarray`.
+
+The following [array data types](https://numpy.org/devdocs/reference/arrays.scalars.html) are supported:
+
+- float32
+- float64
+- int8
+- uint8
+- int16
+- uint16
+- int32
+- uint32
+
+*If the array dtype is **int64** or **uint64**, often the default dtype for arrays in NumPy when no dtype is specified, those dtypes will be changed to other types internally by Plotly.py where possible. When working with NumPY directly, you can [specify the `dtype`](https://numpy.org/doc/stable/user/basics.types.html#array-types-and-conversions-between-types) when creating `ndarray` objects.
+
+Arrays or data types that are not supported for base64 encoding to Plotly.js's typed arrays specification will still work and render correctly with Plotly. Those arrays and or data types just won't have the performance benefits that Plotly.js's base64 typed arrays feature provides.
+
+### Unsupported Attributes
+
+Arrays passed to attributes with the following names are not supported for base64 encoding for rendering with Plotly.js.
+
+`geojson`, `layers`, and `range`.
+
+Attributes that are not supported for base64 encoding to Plotly.js's typed arrays specification will still work and render correctly. Those attributes just won't have the performance benefits that Plotly.js's base64 typed arrays feature provides.
+
+### Example with NumPy Arrays
+
+Here, we use NumPy arrays with a `go.Scatter3d` figure.
+
+```python
+import plotly.graph_objects as go
+import numpy as np
+
+np.random.seed(1)
+
+# Number of data points
+N = 10000
+
+# Generate random data
+x = np.random.randn(N)
+y = np.random.randn(N).astype('float32')
+z = np.random.randint(size=N, low=0, high=256, dtype='uint8')
+c = np.random.randint(size=N, low=-10, high=10, dtype='int8')
+
+fig = go.Figure(data=[go.Scatter3d(
+    x=x,
+    y=y,
+    z=z,
+    marker=dict(color=c),
+    mode='markers',
+    opacity=0.2
+)])
+
+fig.show()
+```
+
+## WebGL
+
+`plotly` figures are rendered by web browsers, which broadly speaking have two families of capabilities for rendering graphics:
+
+- The SVG API, which supports vector rendering.
+- The Canvas API, which supports raster rendering, and can exploit GPU hardware acceleration via a browser technology known as WebGL.
+
+Each `plotly` trace type is rendered with either SVG or WebGL. The following trace types use WebGL for rendering:
+
+* Accelerated versions of SVG trace types: `scattergl`, `scatterpolargl`,
+* High-performance multidimensional trace types: `splom`, or `parcoords`
+* 3D trace types `scatter3d`, `surface`, `mesh3d`, `cone`, `streamtube`
+* Mapbox Gl JS-powered trace types: `scattermapbox`, `choroplethmapbox`, `densitymapbox`
```

Comment:
Do we want to keep mapbox traces here?
If so then we may add a deprecation note?

Reply:
Good point. Swapped out for the new traces